src/taskProvider.ts

```diff
@@ -555,34 +601,70 @@ export class TaskProvider implements vscode.TaskProvider {
   }
 
   private functionalTestsExec(): string {
+    let testsSelection = "";
+    if (this.selectedTests) {
+      testsSelection = "-k \"";
+      // Create list with "or" separator for pytest selection (the last 'or' is not needed)
+      for (let i = 0; i < this.selectedTests.length - 1; i++) {
+        testsSelection += this.selectedTests[i] + " or ";
+      }
+      testsSelection += this.selectedTests[this.selectedTests.length - 1] + "\"";
+    }
     // Runs functional tests inside the docker container (with Qt display disabled).
     const exec = `docker exec -it ${this.containerName} bash -c 'pytest ${
       this.functionalTestsDir
-    } --tb=short -v --device ${this.tgtSelector.getSelectedSpeculosModel()}'`;
+    } --tb=short -v --device ${this.tgtSelector.getSelectedSpeculosModel()} ${testsSelection}'`;
     return exec;
   }
 
   private functionalTestsDisplayExec(): string {
+    let testsSelection = "";
+    if (this.selectedTests) {
+      testsSelection = "-k \"";
+      // Create list with "or" separator for pytest selection (the last 'or' is not needed)
+      for (let i = 0; i < this.selectedTests.length - 1; i++) {
+        testsSelection += this.selectedTests[i] + " or ";
+      }
+      testsSelection += this.selectedTests[this.selectedTests.length - 1] + "\"";
+    }
     // Runs functional tests inside the docker container (with Qt display enabled).
     const exec = `docker exec -it ${this.containerName} bash -c 'pytest ${
       this.functionalTestsDir
-    } --tb=short -v --device ${this.tgtSelector.getSelectedSpeculosModel()} --display'`;
+    } --tb=short -v --device ${this.tgtSelector.getSelectedSpeculosModel()} --display ${testsSelection}'`;
     return exec;
   }
 
   private functionalTestsGoldenRunExec(): string {
+    let testsSelection = "";
```

Comment:
This bloc is written several times and should be factorize in a function like `setSelectedTests`), which returns an empty string of the full filter `"-k ... or ..."`

Reply:
Ok, adding a function.